### DISCUSSION TITLE
Testing the installation: ValueError: numpy.dtype size changed

### DISCUSSION THREAD

**Discussion by siegLoesch** [firedrakeproject/firedrake#3879] [Firedrake support]
I was installing firedrake with the using my system version of PETSc and openmpi using the following options:
`firedrake-install --mpiexec=/opt/ompi/bin/mpiexec --mpicc=/opt/ompi/bin/mpicc --mpicxx=/opt/ompi/bin/mpicxx --mpif90=/opt/ompi/bin/mpif90 --honour-petsc-dir --verbose`
The installation went fine but got an error after the installation test with:
`pytest tests/regression/ -k "poisson_strong or stokes_mini or dg_advection"`

```
ImportError while loading conftest '/home/loesch/.virtualenvs/firedrake/src/firedrake/tests/conftest.py'.
tests/conftest.py:4: in <module>
    from firedrake.petsc import PETSc, get_external_packages
firedrake/__init__.py:33: in <module>
    petsc4py.init(sys.argv)
../../lib/python3.11/site-packages/petsc4py/__init__.py:42: in init
    PETSc = petsc4py.lib.ImportPETSc(arch)
../../lib/python3.11/site-packages/petsc4py/lib/__init__.py:30: in ImportPETSc
    return Import('petsc4py', 'PETSc', path, arch)
../../lib/python3.11/site-packages/petsc4py/lib/__init__.py:97: in Import
    module = import_module(pkg, name, path, arch)
../../lib/python3.11/site-packages/petsc4py/lib/__init__.py:76: in import_module
    spec.loader.exec_module(module)
petsc4py/PETSc.pyx:1: in init petsc4py.PETSc
    ???
E   ValueError: numpy.dtype size changed, may indicate binary incompatibility. Expected 96 from C header, got 88 from PyObject
```

Appreciate any help

Siegfried
[firedrake-install.log](https://github.com/user-attachments/files/17870617/firedrake-install.log)

**Comment by connorjward** [comment]
How old is your system PETSc? I've seen this error before and I *think* that it is caused by the recent-ish numpy v1->v2 change. If you use an updated PETSc then things may work.

As an aside, using a generic system PETSc is not really recommended for a few reasons:
* We actually build and use our own fork (https://github.com/firedrakeproject/petsc). 99% of things will work using regular PETSc but there is one outstanding bug that means we cannot quite use upstream.
* We need a number of specific packages to be installed with PETSc (e.g. MUMPS and HDF5). You should build a version of PETSc using the options from `firedrake-install --show-configure-options`.

**Reply by siegLoesch** [reply]
1) Your hint to numpy versions did the trick: I had to downgrade of numpy :
`pip install --upgrade numpy=="1.26.4"` The tests  succeeded but with lost of warnings which I have to check next. At least I can use firedrake for now - Thank you!
2) The PETSc version in use  on my system is self build: 3.22.1 which is latest tag of main.
3) Reason for linking to my system PETSc was a failure with the default installation (have to analyze that further):
```
=================================================================================
                 Running libtoolize on PNETCDF; this may take several minutes
=================================================================================
=================================================================================
                 Running autoreconf on PNETCDF; this may take several minutes
=================================================================================
=================================================================================
                 Running configure on PNETCDF; this may take several minutes
=================================================================================
************************************************************************************
           UNABLE to CONFIGURE with GIVEN OPTIONS (see configure.log for details):
---------------------------------------------------------------------------------------------
                              Error running configure on PNETCDF
************************************************************************************
2024-11-22 14:26:55,456 DEBUG  Popping path '/home/loesch/installOpt/fem/firedrake/firedrake/src/petsc'
2024-11-22 14:26:55,456 DEBUG  New path '/home/loesch/installOpt/fem/firedrake/firedrake/src'
2024-11-22 14:26:55,458 INFO   
Install log saved in /home/loesch/installOpt/fem/firedrake/firedrake-install.log
```
